### PR TITLE
Packagifying StorageClass

### PR DESCRIPTION
**Pull request by saji-pivotal**
### What this PR does / why we need it
This PR creates the scaffolding needed to carvelize StorageClass being used

### Which issue(s) this PR fixes

Fixes #https://github.com/vmware-tanzu/tanzu-framework/issues/2321

### Describe testing done for PR
1. Used `make package-bundle` and `make push-package-bundles` to build and publish the package to my personal Dockerhub account
2. Ran `kubectl apply -f packages/management/tkg-storageclass/package.yaml` to make the package available on a management-cluster
3. Ran `kubectl apply -f packages/management/tkg-storageclass/metadata.yaml` to make the PackageMetadata available on the management-cluster
4. Ran `tanzu plugin install <package-name>` and verified that the StorageClass was installed on the cluster with the parameters specified

### Release note

```release-note
Creates scaffolding to convert storage class to a carvel package
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: packages/standalone/tkg-storageclass/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  name: tkg-storageclass.tanzu.vmware.com
+  namespace: tanzu-package-repo-global
+spec:
+  displayName: "tkg-storageclass"
+  longDescription: "tkg-storageclass defines the storageclass resources for the cluster"
+  shortDescription: "Defines the storageclass resources for the cluster"
+  providerName: VMware
+  maintainers:
+    - name: ""
```
Comment (tenczar) [inline packages/standalone/tkg-storageclass/metadata.yaml:12]: should you be here, or maybe the vmware-tanzu project on Github?

**Review by tenczar — APPROVED** [review]
Just a few comments, but looks good to me.

File: packages/standalone/tkg-storageclass/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# tkg-storageclass-aws Package
```
[the remark this answers is not in the document]
Comment (saji-pivotal) [reply packages/standalone/tkg-storageclass/README.md:1]: I'm not wholly sure. I just copied over these files from older packages

File: packages/standalone/tkg-storageclass/bundle/config/upstream/base.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:yaml", "yaml")
+#@ load("@ytt:data", "data")
+
+
+#@ if data.values.metadata.infraProvider == "aws":
+---
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (saji-pivotal) [reply packages/standalone/tkg-storageclass/bundle/config/upstream/base.yaml:9]: These contents are identical to the storage class CRs that were getting applied to the cluster by using ytt overlays. The old ytts did not contain `apiVersion`. That's why I've not added them here as well.

File: packages/standalone/tkg-storageclass/metadata.yaml (hunk 1 above)
Comment (saji-pivotal) [reply packages/standalone/tkg-storageclass/metadata.yaml:12]: Good point. I've added my name here

File: packages/standalone/tkg-storageclass/bundle/config/upstream/base.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (saji-pivotal) [reply packages/standalone/tkg-storageclass/bundle/config/upstream/base.yaml:9]: Ah yeah..I just copied over old yaml. I've updated this to have apiVersion on top

**Review by anujc25 — APPROVED** [review]
LGTM

**Pull request merged**